tests/ui/generic-const-items/recursive.stderr

```diff
@@ -0,0 +1,11 @@
+error: queries overflow the depth limit!
+  --> $DIR/recursive.rs:10:13
+   |
+LL |     let _ = RECUR::<()>;
+   |             ^^^^^^^^^^^
+   |
+   = help: consider increasing the recursion limit by adding a `#![recursion_limit = "30"]` attribute to your crate (`recursive`)
+   = note: query depth increased by 17 when simplifying constant for the type system `RECUR`
```

Comment:
Would be neat if we could report the query backtrace before that recursion

Reply:
(Similar to cycle errors)

Reply:
I'm not exactly sure how this should work since the `job.id` never repeats in this case. I guess we could determine when the cycle is complete by finding the first parent query with `job.dep_kind == eval_to_const_value_raw` and a span that matches the one of the current query. What do you think about this?